### PR TITLE
fix(`synced-lyrics`): Fix 2 issues

### PR DESCRIPTION
**Pull request by Reycko**
First, this PR fixes an issue with the default text string never showing, as it verified whether it was defined, rather than whether or not it's empty.

Secondly, this fixes another, more important issue.
[Line 182 of the fetcher](https://github.com/th-ch/youtube-music/blob/f2716e1dc8c296dfb6a6d7dfc586c9557fca9a1e/src/plugins/synced-lyrics/renderer/lyrics/fetch.ts#L182) uses the `index` element of the `reduce` function, however, this would cause every lyric after an unparsed one to be offset by 1, causing miscalculations for `duration` thus multiple lines showing as current. This PR fixes the issue by using an iterator.

Before:
image

After:
image

### CODE REVIEW

**Comment by Dzheremi2** [comment]
If you fixing issue with default string value, please make the ability to disable it to make the same effect as it was with this bug.

PLEASE! This effect, without default string value are beautiful!

**Comment by Reycko** [comment]
> If you fixing issue with default string value, please make the ability to disable it to make the same effect as it was with this bug.
> 
> PLEASE! This effect, without default string value is beautiful!

There's already a ` ` default value which should do the same effect

File: src/plugins/synced-lyrics/renderer/lyrics/fetch.ts [hunk 1]
```diff
@@ -178,10 +178,12 @@ export const getLyricsList = async (
   // Add a blank line at the beginning
   raw.unshift('[0:0.0] ');
 
-  const syncedLyricList = raw.reduce<LineLyrics[]>((acc, line, index) => {
-    const syncedLine = extractTimeAndText(line, index);
+  let i = 0;
```
Comment (JellyBrick) [inline src/plugins/synced-lyrics/renderer/lyrics/fetch.ts]: Why did you change this line?

**Review by JellyBrick — CHANGES_REQUESTED** [review]

File: src/plugins/synced-lyrics/renderer/lyrics/fetch.ts (hunk 1 above)
Comment (Reycko) [reply src/plugins/synced-lyrics/renderer/lyrics/fetch.ts]: raw.reduce doesn't account for `acc`'s length. If we don't push a line, `index` would still be increased by 1.
